lighthouse-core/lib/page-functions.js

```diff
@@ -453,12 +452,13 @@ const getNodeDetailsString = `function getNodeDetails(elem) {
   ${getBoundingClientRect.toString()};
   ${getOuterHTMLSnippet.toString()};
   ${getNodeLabel.toString()};
+  const shadowRootElem = elem instanceof ShadowRoot ? elem.host : elem;
```

Comment:
nit: the value of `shadowRootElem` will not necessarily be a shadow root. I think something like `htmlElem` makes more sense.